pkg/cloudprovider/aws/instancetype.go

```diff
@@ -70,10 +70,10 @@ func (i *InstanceType) Memory() *resource.Quantity {
 }
 
 func (i *InstanceType) Pods() *resource.Quantity {
-	// The number of pods per node is calculated using the formula:
-	// max number of ENIs * (IPv4 Addresses per ENI -1) + 2
-	// https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt#L20
-	return resources.Quantity(fmt.Sprint(*i.NetworkInfo.MaximumNetworkInterfaces*(*i.NetworkInfo.Ipv4AddressesPerInterface-1) + 2))
+	if i.MaxPods != nil {
+		return resources.Quantity(fmt.Sprint(*i.MaxPods))
```

Comment:
consider ptr.Int32Value, which we frequently use as an insurance policy against crashing.

Reply:
I went this route because Quantity needs a string and I saw it used in several of the other places Quantity was being used. May be a chance to loop back and clean all of them up in the future?

Reply:
Ah... I get what you mean now. Fixing it 👍